### PR TITLE
Add logs

### PR DESCRIPTION
**Pull request by kdada**
1. add log for http request

### CODE REVIEW

File: pkg/lock/resource.go [hunk 1]
```diff
@@ -116,17 +149,22 @@ func NewResourceLock(creator func() Locker) *ResourceLock {
...
-	for _, r := range res {
+	for i, r := range res {
 		lock, ok := children[r]
 		if !ok {
 			lock = NewHierarchicalLock(rl.CreateLock())
 			children[r] = lock
 			children = lock.Children
 		}
-		result = append(result, lock.Lock)
+		result.locks[i] = lock.Lock
```
Comment (ddysher) [inline pkg/lock/resource.go]: This is not "Add log", please do this in another PR.

File: pkg/lock/resource.go [hunk 2]
```diff
@@ -22,13 +27,27 @@ func (dl deadline) left() time.Duration {
...
+
+// Name returns name of locks
+func (l *Locks) Name() string {
+	return fmt.Sprintf("%s(%d)", l.name, l.id)
+}
 
 // Lock tries lock for writing. If locked, return true
-func (locks Locks) Lock(timeout time.Duration) bool {
+func (l *Locks) Lock(timeout time.Duration) bool {
+	log.Infof("lock %s", l.Name())
```
Comment (ddysher) [inline pkg/lock/resource.go]: nit: Is this too verbose?  Can we increase log level?

Comment (kdada) [inline pkg/lock/resource.go]: Log package does not support log level. And `Info` is low-level log.

**Comment by kdada** [comment]
@ddysher  PTAL
